mailpoet/assets/js/src/context/use-notices.jsx

```diff
@@ -46,12 +48,69 @@ export const useNotices = () => {
     [add],
   );
 
+  /**
+   * This method fundamentally performs the same function as useNotices.error
+   * The only addition is the wrapper for checking reinstall_plugin error
+   */
+  const apiError = useCallback(
+    (response, props = {}) => {
+      let content = response;
+      let optionsObj = props;
+
+      if (response && response.errors && response.errors.length > 0) {
+        const containsReinstallPluginMessage = JSON.stringify(
+          response.errors,
+        ).includes('reinstall_plugin');
+
+        if (containsReinstallPluginMessage) {
+          content = response.errors.map((err) => (
+            <p key={err.message}>
+              {err.error === 'reinstall_plugin'
+                ? createInterpolateElement(
+                    sprintf(
+                      __(
+                        'The plugin has encountered an unexpected error. Please reload the page. If that does not help, re-install the MailPoet Plugin. See: %s for more information',
+                        'mailpoet',
+                      ),
+                      '<a> https://kb.mailpoet.com/article/258-re-installing-updating-the-plugin-via-ftp </a>',
```

Comment:
> The text in the link element should be different from the ticket specification. Or is there a reason why you used this approach?

Uh? Why should the text in the link element be different?

> Also, we usually use the mark here, which can be a part of the translation string, and translators already know it. Maybe we should stay with this approach.

I'm lost. Are you suggesting we use <mark> instead of <a> ?  We need the <a> for the interpolation.

> Is there some reason for the redundant spaces around the text inside the tag?

Not really. It can/will be removed.

Reply:
> Uh? Why should the text in the link element be different?
> I'm lost. Are you suggesting we use instead of ? We need the for the interpolation.

@triple0t The ticket specification contains the following sentence:
`The plugin has encountered an unexpected error. Please reload the page. If that doesn't help [link]re-install the MailPoet Plugin.[/link]`,
but you used: `The plugin has encountered an unexpected error. Please reload the page. If that does not help, re-install the MailPoet Plugin. See: %s for more information`.
Displaying the target URL can make the text less readable. I don't think we use this approach anywhere.
Also, when we use string replacing via `sprintf` with the `%s` specifier, we should add a comment for translators. [Here](https://github.com/mailpoet/mailpoet/blob/c143da4ccfe266beb6091ce0ede742bad2ff639c/mailpoet/assets/js/src/automation/listing/components/menu/delete.tsx#L47-L51) is an example.

Nevertheless, If I add this message, I would try to follow the ticket specification exactly, and it could look like this:
```
createInterpolateElement(
  __(
    'The plugin has encountered an unexpected error. Please reload the page. If that does not help, <link>re-install the MailPoet Plugin.</link>,
    'mailpoet',
  ),
  {
    link: (
    <a
      href="https://kb.mailpoet.com/article/258-re-installing-updating-the-plugin-via-ftp"
      target="_blank"
      rel="noopener noreferrer"
    >
      &nbsp;
    </a>
    ),
  },
)
```

Maybe we can display the whole URL in the text, but in my opinion, it should be at the end of the sentence and not in the middle, and we would need to add a comment for translators.

Reply:
Noted. 
Fixed here https://github.com/mailpoet/mailpoet/pull/5592/commits/656a66299bd1b2c85449b60b394983bdfab88455
Thanks